### ISSUE TITLE
Dependency cycle when using supervisord::group 

### ISSUE THREAD

**Issue by dol** [ajcrowe/puppet-supervisord#79]
The latest release 0.6.0 (workes with 0.5.1) has a problem with a dependency cycle when using the following example:

```puppet
include ::supervisord

supervisord::program { 'program1':
  command             => 'command --args',
}

supervisord::program { 'program2':
  command             => 'command --args',
}

supervisord::group { 'mygroup':
  programs => ['program1', 'program2']
}
```

Error message:
```
==> default: Error: Failed to apply catalog: Found 1 dependency cycle:
==> default: (Exec[supervisorctl_command_start_program1] => Supervisord::Supervisorctl[start_program1] => Supervisord::Program[program1] => Supervisord::Group[mygroup] => File[/etc/supervisor.d/group_mygroup.conf] => Class[Supervisord::Reload] => Exec[supervisorctl_update] => Class[Supervisord::Reload] => Supervisord::Supervisorctl[start_program1] => Exec[supervisorctl_command_start_program1])
```

To visualize the problem I created a dependency graph.

![image](http://i.imgur.com/CO370Zx.png).

**Comment by ZhukV** [comment]
+1

**Comment by M4t3o** [comment]
+1

**Comment by ajcrowe** [comment]
Will try and look at this soon, if you've got a working solution please do submit a PR even if incomplete.

**Comment by Vincent--** [comment]
See https://github.com/RewardGateway/puppet-supervisord/commit/06211c7e9005ce29faa4e3707052cf1c149e7340

These dependencies are not necessary:
```
Supervisord::Program <| |> -> Supervisord::Group <| |>
Supervisord::Fcgi_program <| |> -> Supervisord::Group <| |>
Supervisord::Eventlistener <| |> -> Supervisord::Group <| |>
```

We only need to ensure that conf files for  group and programs are present before reloading supervisor...

**Comment by ajcrowe** [comment]
Closing as fixed in #93
